Sources/MapboxNavigation/Map.swift

```diff
@@ -0,0 +1,30 @@
+
+import Foundation
+import MapboxCoreMaps
+
+extension Map {
+    func getStyleSourceURLs(_ sourceTypes: [String]) -> [String] {
+        let filteredSources = try! getStyleSources().filter {
+            return sourceTypes.contains($0.type)
+        }
+        
+        var urls = [String]()
+        for source in filteredSources {
+            let properties = try! getStyleSourceProperties(forSourceId: source.id)
+            
+            if properties.isValue() {
+                let contents = properties.value as? [String: AnyObject]
+                guard var url = contents?["url"] as? String else {
+                    continue
+                }
+                if url.split(separator: ",").count > 1 || !url.hasPrefix("mapbox://") {
+                    continue // we ignore composite (https://docs.mapbox.com/studio-manual/reference/styles/#source-compositing.) and non-mapbox sources
+                } else {
+                    url.removeFirst("mapbox://".count)
+                    urls.append(url)
+                }
```

Comment:
`URL` provides more robust URL parsing, for example:

https://github.com/mapbox/mapbox-navigation-ios/blob/a222d2d8bd5d7f9a6b9a0a19746026034314394a/Sources/MapboxNavigation/MGLVectorTileSource.swift#L10-L15